lib/Analysis/Allocation.cpp

```diff
@@ -66,6 +66,9 @@ SmallVector<unsigned> getRepShapeForCvtLayout(triton::gpu::ConvertLayoutOp op) {
     return convertType<unsigned, int64_t>(getShapePerCTA(srcTy));
   }
 
+  if (isMfmaToDotShortcut(srcTy, dstTy))
```

Comment:
In the implementation of `isMfmaToDotShortcut()`, we assume the src and dst tensor have mfma and dot layouts. Here we need to guarantee the assumption holds.
I'm wondering how this does not break anything when src is not mfma or dst is not dot layout.